src/makeCalculator.js

```diff
@@ -1,10 +1,41 @@
 'use strict';
 
 /**
+ * Generates an object with main calculator operations.
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(operation, value) {
+      this.result = operation(value, this.result);
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(value, result) {
+      return result + value;
+    },
+
+    subtract(value, result) {
+      return result - value;
+    },
+
+    divide(value, result) {
```

Comment:
what happens if i pass 0 as value?)

Reply:
`this.result `will be Infinity.

There was no any requirements about 'dividing by zero' case handling. Should I reset result to 0 or throw an error or something else?

Reply:
> `this.result `will be Infinity.
> 
> There was no any requirements about 'dividing by zero' case handling. Should I reset result to 0 or throw an error or something else?

Ok, it's just for a better user experience. Yes, you can throw an error or just notify a user that they can't divide by 0